### ISSUE TITLE
Reproducibility

### ISSUE THREAD

**Issue by zhykoties** [jsyoon0823/TimeGAN#29]
Could you share the hyperparameters that are needed to achieve the desired performance as listed in Table 2 of the paper? I got similar numbers for the discriminator to what Atrin78 mentioned here (https://github.com/jsyoon0823/TimeGAN/issues/7) by using the default hyperparameters.

--data_name stock --seq_len 24 --module gru --hidden_dim 24 --num_layer 3 --iteration 10000 --batch_size 128 --metric_iteration 10
{'discriminative': 0.20225102319236016, 'predictive': 0.03779167104371475}
--data_name stock --seq_len 24 --module gru --hidden_dim 24 --num_layer 3 --iteration 50000 --batch_size 128 --metric_iteration 10
{'discriminative': 0.12578444747612552, 'predictive': 0.03670826588897788}
--data_name energy --seq_len 24 --module gru --hidden_dim 24 --num_layer 3 --iteration 50000 --batch_size 128 --metric_iteration 10
{'discriminative': 0.4985924423028151, 'predictive': 0.32724065483352016}

There are also some discrepancies between the hyperparameters used in the code and the hyperparameters mentioned in the paper. I roughly tried both versions and some of their combinations but was still not able to achieve the mentioned performance. Using the hyperparameters in the paper I got:
Stock: 
run 1:
discriminative: 0.212; predictive: 0.039
run 2:
discriminative: 0.163; predictive: 0.040

Could you kindly share all the hyperparameters that are needed to get the performance in the paper (stock: .102, 0.038; energy: .236, .273)? 

Additionally, as mentioned in https://github.com/jsyoon0823/TimeGAN/issues/25, in the second stage of training, we have GS_solver = tf.train.AdamOptimizer().minimize(G_loss_S, var_list=g_vars + s_vars). However, G_loss_S does not seem to depend on g_vars and thus only s_vars are updated during this stage of training. So should it be GS_solver = tf.train.AdamOptimizer().minimize(G_loss_S, var_list=s_vars) only? Or is there something else I am missing? Based on tutorial_timegan.ipynb, in the second stage of training, the printed numbers on this Github repo are
Start Training with Supervised Loss Only
step: 0/10000, s_loss: 0.273
step: 1000/10000, s_loss: 0.0191
step: 2000/10000, s_loss: 0.009
step: 3000/10000, s_loss: 0.0071
step: 4000/10000, s_loss: 0.0054
step: 5000/10000, s_loss: 0.0054
step: 6000/10000, s_loss: 0.0042
step: 7000/10000, s_loss: 0.0041
step: 8000/10000, s_loss: 0.003
step: 9000/10000, s_loss: 0.0028

s_loss keeps decreasing and reaches 0.0028. However, when I tried running the code locally with all the hyperparameters being set to the same values, I get the following s_loss in the second stage:
step: 1000/10000, s_loss: 0.0263
step: 2000/10000, s_loss: 0.0234
step: 3000/10000, s_loss: 0.0222
step: 4000/10000, s_loss: 0.0205
step: 5000/10000, s_loss: 0.0201
step: 6000/10000, s_loss: 0.0187
step: 7000/10000, s_loss: 0.0202
step: 8000/10000, s_loss: 0.0196
step: 9000/10000, s_loss: 0.0196

s_loss seems to stay around 0.02 instead of 0.002 as in the uploaded notebook results. Could you confirm if there's anything I might be missing?

Thanks in advance!

**Comment by jsyoon0823** [comment]
Thank you for your interest in our paper.

(1) Unfortunately, I did not save the hyper-parameters for all the datasets.
- However, the current hyper-parameters in the main_timegan.py work well
(and reproduce) the STOCK result.
- For energy, maybe you need to increase the hidden dimension size.

(2) Supervised
- You are correct. You can change the var_list as you mentioned.

(3) s_loss
- In my case, I can reproduce the same s_loss in a jupyter notebook.

Thanks!
